tests/MongoDB.Driver.Tests/Specifications/change-streams/ChangeStreamTestRunner.cs

```diff
@@ -301,6 +302,20 @@ private List<ChangeStreamDocument<BsonDocument>> ReadChangeStreamDocuments(IAsyn
             }
 
             return result;
+
+            IEnumerable<IEnumerable<ChangeStreamDocument<BsonDocument>>> ReliableMoveNext()
+            {
+                var stopwatch = Stopwatch.StartNew();
+                while (async ? cursor.MoveNextAsync().GetAwaiter().GetResult() : cursor.MoveNext())
+                {
+                    if (cursor.Current.Any() || stopwatch.Elapsed > TimeSpan.FromSeconds(10))
```

Comment:
not sure about 10 seconds, maybe we need to find less timeout value, nut maybe it's a good enough value for tests